File: RumahSehat/src/main/java/apap/tugas/akhir/RumahSehat/rest/ResepDetail.java
package apap.tugas.akhir.RumahSehat.rest;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Getter;
import lombok.Setter;

@Getter @Setter
@JsonIgnoreProperties(ignoreUnknown = true)
public class ResepDetail {
    @JsonProperty("id")
    private String id;

    @JsonProperty("nama_dokter")
    private String namaDokter;

    @JsonProperty("nama_pasien")
    private String namaPasien;

    @JsonProperty("is_done")
    private String isDone;

    @JsonProperty("nama_apoteker")
    private String namaApoteker;
    
    @JsonProperty("list_jumlah")
    private List<String> listJumlah;
}

File: rumahsehat_flutter/lib/screens/resep/resep_detail.dart
class DetailResep {
  String id;
  String namaDokter;
  String namaPasien;
  String isDone;
  String namaApoteker;
  List<String> listJumlah;

  DetailResep(
    {required this.id,
      required this.namaDokter,
      required this.namaPasien,
      required this.isDone,
      required this.namaApoteker,
      required this.listJumlah
    }
  );

  factory DetailResep.fromJson(Map<String, dynamic> json) {
    return DetailResep(
      id: json['id'],
      namaDokter: json['nama_dokter'],
      namaPasien: json['nama_pasien'],
      isDone: json['is_done'],
      namaApoteker: json['nama_apoteker'],
      listJumlah: json['list_jumlah'],
    );
  }
}
